### PR TITLE
Dịch "environment_vn" - Phần 7

### LINKED ISSUE #998: Dịch "environment_vn" - Phần 7
Phần cần dịch: https://github.com/aivivn/d2l-vn/blame/master/chapter_multilayer-perceptrons/environment_vn.md#L390-L456

### Hướng dẫn
- [Dịch trực tiếp trên website](https://github.com/aivivn/d2l-vn/blob/master/CONTRIBUTING.md).
- Sau khi nhấn vào phần cần dịch ở trên, hãy chọn **Normal view** và làm theo hướng dẫn ở trên để có thể bắt đầu dịch.

### PR DESCRIPTION
**Pull request by nguyenduydu**
Phần này mình thấy khá khó đọc và khó hiểu. Mặc dù đã dành nhiều thời gian hơn so với các phần khác nhưng nhiều đoạn chưa biết dịch sao cho hợp lý hơn. Mong mọi người góp ý.

## Pull Request Checklist

**Đánh dấu tick (sau khi "Create Pull Request") vào các mục dưới đây:**

* [x] Pull Request này tương ứng với issue nào? Trả lời số issue sau dấu `#` (không có dấu cách): Close #998

* [x] Bản dịch này có bám sát **[Bảng thuật ngữ](https://github.com/aivivn/d2l-vn/blob/master/glossary.md)** không? Nếu một từ/cụm chưa có trong Bảng thuật ngữ, hãy tạo một Pull Request riêng đề xuất cách dịch từ/cụm đó.

* [x] Format của bản gốc có được giữ nguyên không?
    * Các phần in nghiêng, in đậm.
    * Tiêu đề (số lượng dấu `#` đầu dòng).
    * Bảng biểu, chú thích cho bảng (dòng phía trên mỗi bảng bắt đầu bằng dấu `:`).
    * Dịch các từ trong hình vẽ nếu cần. Các dòng có hình có dạng: `caption`.
    * Dịch các chú thích hình vẽ (thay các cụm `*dịch chú thích ảnh phía trên*` bằng bản dịch tiếng Việt).
    * Không dịch các phần code (nằm giữa hai dấu `````).
    * Copy các công thức toán từ bản gốc (các đoạn có `$`).
    * Giữ các dòng gán nhãn (bắt đầu với `:label:`).
    * Không tự thêm bớt các dòng trắng.
     
* [x] Trong một đoạn văn, mỗi câu văn đã được viết trong một dòng, giữa các dòng không có dòng trắng. Markdown sẽ coi những dòng liền nhau không có dòng trắng là một đoạn văn. Việc này giúp công đoạn review được thuận tiện hơn.

* [x] Tất cả chữ trong các hình vẽ đã được dịch. Nếu chưa, vui lòng xem thêm **[hướng dẫn hoặc yêu cầu dịch hình](https://github.com/aivivn/d2l-vn/blob/master/img/README.md)**.

* [x] Điền tên của bạn vào mục "Những người thực hiện" ở cuối file.

* [ ] **Sau khi được approve**, điền tên những reviewers mà bạn thấy có nhiều đóng góp cho Pull Request này. Xem danh sách Contributors **[tại đây](https://github.com/aivivn/d2l-vn/blob/master/docs/contributors_info.md)**.

## Với Reviewer

1. Chỉ review nếu Pull Request này đã sẵn sàng cho việc review.

2. Đã đọc kỹ hướng dẫn đóng góp và xem qua **[Bảng thuật ngữ](https://github.com/aivivn/d2l-vn/blob/master/glossary.md)**.

3. **Bạn có thể đề xuất thay đổi cách dịch mỗi dòng trực tiếp như trong hình dưới đây:**
img

### CODE REVIEW

File: chapter_multilayer-perceptrons/environment_vn.md [hunk 1]
```diff
@@ -404,23 +406,31 @@ Here's a prototypical algorithm for that purpose which uses an unlabeled trainin
...
+Nếu một điểm có $q(\mathbf{x}) > 0$ nhưng $p(\mathbf{x}) = 0$, thì trọng số quan trọng tương ứng bằng vô hạn.
 
 <!--
 *Generative Adversarial Networks* use a very similar idea to that described above to engineer a *data generator* that outputs data that cannot be distinguished from examples sampled from a reference dataset.
 In these approaches, we use one network, $f$ to distinguish real versus fake data and a second network $g$ that tries to fool the discriminator $f$ into accepting fake data as real.
 We will discuss this in much more detail later.
 -->
 
-*dịch đoạn phía trên*
+*Mạng Đối Sinh* sử dụng một ý tưởng rất giống với mô tả ở trên để thiết kế một *bộ tạo dữ liệu* để sinh dữ liệu không thể phân biệt được với các mẫu được lấy từ một bộ dữ liệu tham chiếu.
```
Comment (minhduc0711) [inline chapter_multilayer-perceptrons/environment_vn.md]: mình sửa cho đỡ lặp từ
```suggestion
*Mạng Đối Sinh* sử dụng một ý tưởng rất giống với mô tả ở trên để thiết kế một *bộ tạo dữ liệu* có khả năng sinh dữ liệu không thể phân biệt được với các mẫu được lấy từ một tập dữ liệu tham chiếu.
```

File: chapter_multilayer-perceptrons/environment_vn.md [hunk 2]
```diff
@@ -442,7 +452,9 @@ then we can get consistent estimates of these weights without ever having to dea
 while the labels are often easier to work, say vectors whose length corresponds to the number of classes).
 -->
 
-*dịch đoạn phía trên*
+Để thảo luận về dịch chuyển nhãn, ta sẽ giả định rằng ta đang giải quyết một bài toán phân loại $k$ lớp.
+Nếu phân phối của nhãn thay đổi theo thời gian $p(y) \neq q(y)$ nhưng các phân phối có điều kiện của lớp vẫn giữ nguyên $p(y) \neq q(y)$, thì trọng số quan trọng sẽ tương ứng với tỉ lệ xác suất của nhãn $q(y)/p(y)$.
```
Comment (minhduc0711) [inline chapter_multilayer-perceptrons/environment_vn.md]: nhầm công thức nè, `likelihood` thì mình thấy mọi người hay dịch là `(độ) hợp lý`
```suggestion
Nếu phân phối của nhãn thay đổi theo thời gian $p(y) \neq q(y)$ nhưng các phân phối có điều kiện của lớp vẫn giữ nguyên $p(\mathbf{x})=q(\mathbf{x})$, thì trọng số quan trọng sẽ tương ứng với tỉ lệ hợp lý của nhãn $q(y)/p(y)$.
```

File: chapter_multilayer-perceptrons/environment_vn.md [hunk 3]
```diff
@@ -442,7 +452,9 @@ then we can get consistent estimates of these weights without ever having to dea
...
+Một điều khá hay về dịch chuyển nhãn là nếu sở hữu một mô hình tốt (trên phân phối nguồn) thì ta có thể có được ước lượng phù hợp cho các trọng số này mà không phải đối phó với vấn đề nhiều chiều (trong học sâu, đầu vào thường là dữ liệu nhiều chiều như hình ảnh, trong khi các nhãn thường dễ làm việc hơn vì chỉ là các vector có chiều dài tương ứng với số lượng lớp).
```
Comment (minhduc0711) [inline chapter_multilayer-perceptrons/environment_vn.md]:
```suggestion
Một điều khá hay về dịch chuyển nhãn là nếu ta có một mô hình tương đối tốt (trên phân phối gốc), ta có thể có các ước lượng nhất quán cho các trọng số này mà không phải đối phó với vấn đề nhiều chiều (trong học sâu, đầu vào thường là dữ liệu nhiều chiều như hình ảnh, trong khi các nhãn thường dễ làm việc hơn vì chúng chỉ là các vector có chiều dài tương ứng với số lượng lớp).
```

File: chapter_multilayer-perceptrons/environment_vn.md [hunk 4]
```diff
@@ -451,7 +463,9 @@ The confusion matrix C, is simply a $k \times k$ matrix where each column corres
 Each cell's value $c_{ij}$ is the fraction of predictions where the true label was $j$ *and* our model predicted $y$.
 -->
 
-*dịch đoạn phía trên*
+Để ước lượng phân phối của nhãn mục tiêu, đầu tiên ta dùng một bộ phân loại tương đối tốt (thường được học trên tập huấn luyện) và sử dụng một tập kiểm định (cùng phân phối với tập huấn luyện) để tính ma trận confusion.
```
Comment (minhduc0711) [inline chapter_multilayer-perceptrons/environment_vn.md]: `confusion matrix` tạm block bởi #1022  
```suggestion
Để ước lượng phân phối nhãn mục tiêu, đầu tiên ta dùng một bộ phân loại sẵn có tương đối tốt (thường được học trên tập huấn luyện) và sử dụng một tập kiểm định (cùng phân phối với tập huấn luyện) để tính ma trận confusion.
```

File: chapter_multilayer-perceptrons/environment_vn.md [hunk 5]
```diff
@@ -451,7 +463,9 @@ The confusion matrix C, is simply a $k \times k$ matrix where each column corres
...
+Ma trận confusion C là một ma trận $k \times k$, trong đó mỗi cột tương ứng với một nhãn *thật* và mỗi dòng tương ứng với nhãn dự đoán của mô hình.
```
Comment (minhduc0711) [inline chapter_multilayer-perceptrons/environment_vn.md]:
```suggestion
Ma trận confusion C là một ma trận $k \times k$, trong đó mỗi cột tương ứng với một nhãn *thật* và mỗi hàng tương ứng với nhãn dự đoán của mô hình.
```

File: chapter_multilayer-perceptrons/environment_vn.md [hunk 6]
```diff
@@ -451,7 +463,9 @@ The confusion matrix C, is simply a $k \times k$ matrix where each column corres
...
+Còn giá trị của mỗi phần tử $c_{ij}$ là tỉ lệ dự đoán với nhãn thật là $j$ *và* nhãn dự đoán là $y$.
```
Comment (minhduc0711) [inline chapter_multilayer-perceptrons/environment_vn.md]: tác giả lại nhầm rồi, lại còn tự nhiên in nghiêng từ `and` nữa chứ :)))
```suggestion
Giá trị của mỗi phần tử $c_{ij}$ là tỉ lệ mẫu có nhãn thật là $j$ *và* nhãn dự đoán là $i$.
```

File: chapter_multilayer-perceptrons/environment_vn.md (hunk 2 above)
Comment (nguyenduydu) [reply chapter_multilayer-perceptrons/environment_vn.md]: Mình cũng thấy trong bảng glossary dịch là `hợp lý` nhưng không nhớ sao vẫn lấy là `xác suất`. Chắc do sơ xuất. Cám ơn bạn nhé!

File: chapter_multilayer-perceptrons/environment_vn.md (hunk 6 above)
Comment (nguyenduydu) [reply chapter_multilayer-perceptrons/environment_vn.md]: Đoạn này mình thấy để từ `Còn `ở đầu để nối tiếp câu bên trên vì để không thì khá rời rạc. Có thể đây là ý kiến cá nhân thôi.

Trong bản gốc của sách mình thấy có in nghiêng từ `and`, có thể hàm ý nhấn mạnh quan hệ `và` chăng?

`Each cell's value $c_{ij}$ is the fraction of predictions where the true label was $j$ *and* our model predicted $y$.`

File: chapter_multilayer-perceptrons/environment_vn.md (hunk 4 above)
Comment (nguyenduydu) [reply chapter_multilayer-perceptrons/environment_vn.md]: Đoạn này ban đầu mình cũng dịch `off the shelf` là `sẵn có` nhưng sợ sẽ gây confused cho người đọc vì có thể hiểu thành đây là một bộ phân loại đã có sẵn, tức không cần phải huấn luyện.

File: chapter_multilayer-perceptrons/environment_vn.md (hunk 3 above)
Comment (nguyenduydu) [reply chapter_multilayer-perceptrons/environment_vn.md]: Đoạn này dùng từ `có` thì hợp lý hơn nhưng mình thấy từ này bị lặp lại trong cùng một câu nêu muốn thay bằng từ `sở hữu`.

**Comment by nguyenduydu** [comment]
@minhduc0711  Cám ơn bạn đã góp ý nhé. Sửa lại thấy tốt hơn nhiều rồi :D

File: chapter_multilayer-perceptrons/environment_vn.md [hunk 7]
```diff
@@ -395,7 +395,9 @@ and then a reweighted minimization problem where we weigh terms by $\beta$, e.g.
 Here's a prototypical algorithm for that purpose which uses an unlabeled training set $X$ and test set $Z$:
 -->
 
-*dịch đoạn phía trên*
+Hệ quả là có hai vấn đề cần được giải quyết: đầu tiên là phân biệt giữa dữ liệu được lấy ra từ hai phân phối,
+và sau đó là bài toán tối thiểu hóa với trọng số được đánh lại, trong đó ta đánh trọng số cho các mẫu với $\beta$, ví dụ như thông qua các gradient đầu.
```
[the remark this answers is not in the document]
Comment (nguyenduydu) [reply chapter_multilayer-perceptrons/environment_vn.md]: Đoạn này mình cũng thấy khó hiểu. Theo mình nghĩ thì đoạn này nói về bài toán tối thiểu hóa hàm mất mát với trọng số được đánh lại, còn gradient ở đây chắc đang nói về việc sử dụng gradient descent để giải bài toán này. Nhưng các `head gradient` liên quan thế nào thì mình chưa hiểu, chắc phải nhờ bạn nào hiểu rõ hơn giải thích cho :)

Mình đã xem bài giảng của tác giả (https://www.youtube.com/watch?v=24YZ_62Jn-o)
Nhưng tiếc là không thấy nhắc gì đến chi tiết này.

File: chapter_multilayer-perceptrons/environment_vn.md [hunk 8]
```diff
@@ -430,7 +440,7 @@ We will discuss this in much more detail later.
 ### Label Shift Correction
 -->
 
-### *dịch tiêu đề phía trên*
+### Điều chỉnh Dịch chuyển nhãn
```
Comment (lkhphuc) [inline chapter_multilayer-perceptrons/environment_vn.md]:
```suggestion
### Hiệu chỉnh Dịch chuyển nhãn
```
điều chỉnh chỉ có nghĩa thay đổi, hiệu chỉnh có ý làm đúng lại.

File: chapter_multilayer-perceptrons/environment_vn.md [hunk 9]
```diff
@@ -442,7 +452,9 @@ then we can get consistent estimates of these weights without ever having to dea
 while the labels are often easier to work, say vectors whose length corresponds to the number of classes).
 -->
 
-*dịch đoạn phía trên*
+Để thảo luận về dịch chuyển nhãn, ta sẽ giả định rằng ta đang giải quyết một bài toán phân loại $k$ lớp.
+Nếu phân phối của nhãn thay đổi theo thời gian $p(y) \neq q(y)$ nhưng các phân phối có điều kiện của lớp vẫn giữ nguyên $p(\mathbf{x})=q(\mathbf{x})$, thì trọng số quan trọng sẽ tương ứng với tỉ lệ hợp lý của nhãn $q(y)/p(y)$.
```
Comment (lkhphuc) [inline chapter_multilayer-perceptrons/environment_vn.md]:
```suggestion
Nếu phân phối của nhãn thay đổi theo thời gian $p(y) \neq q(y)$ nhưng các phân phối có điều kiện của lớp vẫn giữ nguyên $p(\mathbf{x})=q(\mathbf{x})$, thì trọng số quan trọng sẽ tương ứng với tỉ lệ hợp lý (*likelihood ratio*) của nhãn $q(y)/p(y)$.
```

File: chapter_multilayer-perceptrons/environment_vn.md [hunk 10]
```diff
@@ -442,7 +452,9 @@ then we can get consistent estimates of these weights without ever having to dea
...
+Một điều khá hay về dịch chuyển nhãn là nếu ta có một mô hình tương đối tốt (trên phân phối gốc), ta có thể có các ước lượng nhất quán cho các trọng số này mà không phải đối phó với vấn đề nhiều chiều (trong học sâu, đầu vào thường là dữ liệu nhiều chiều như hình ảnh, trong khi các nhãn thường dễ làm việc hơn vì chúng chỉ là các vector có chiều dài tương ứng với số lượng lớp).
```
Comment (lkhphuc) [inline chapter_multilayer-perceptrons/environment_vn.md]:
```suggestion
Một điều tốt về dịch chuyển nhãn là nếu ta có một mô hình tương đối tốt (trên phân phối gốc), ta có thể có các ước lượng nhất quán cho các trọng số này mà không phải đối phó với không gian đầu vào (trong học sâu, đầu vào thường là dữ liệu nhiều chiều như hình ảnh), trong khi các nhãn thường dễ làm việc hơn vì chúng chỉ là các vector có chiều dài tương ứng với số lượng lớp).
```

File: chapter_multilayer-perceptrons/environment_vn.md [hunk 11]
```diff
@@ -451,7 +463,9 @@ The confusion matrix C, is simply a $k \times k$ matrix where each column corres
 Each cell's value $c_{ij}$ is the fraction of predictions where the true label was $j$ *and* our model predicted $y$.
 -->
 
-*dịch đoạn phía trên*
+Để ước lượng phân phối nhãn mục tiêu, đầu tiên ta dùng một bộ phân loại sẵn có tương đối tốt (thường được học trên tập huấn luyện) và sử dụng một tập kiểm định (cùng phân phối với tập huấn luyện) để tính ma trận confusion.
```
Comment (lkhphuc) [inline chapter_multilayer-perceptrons/environment_vn.md]:
```suggestion
Để ước lượng phân phối nhãn mục tiêu, đầu tiên ta dùng một bộ phân loại sẵn có tương đối tốt (thường được học trên tập huấn luyện) và sử dụng một tập kiểm định (cùng phân phối với tập huấn luyện) để tính ma trận nhầm lẫn.
```
Chắc từ này sẽ được merge, gợi ý trước :).

File: chapter_multilayer-perceptrons/environment_vn.md [hunk 12]
```diff
@@ -451,7 +463,9 @@ The confusion matrix C, is simply a $k \times k$ matrix where each column corres
...
+Ma trận confusion C là một ma trận $k \times k$, trong đó mỗi cột tương ứng với một nhãn *thật* và mỗi hàng tương ứng với nhãn dự đoán của mô hình.
```
Comment (lkhphuc) [inline chapter_multilayer-perceptrons/environment_vn.md]:
```suggestion
Ma trận nhầm lẫn C là một ma trận $k \times k$, trong đó mỗi cột tương ứng với một nhãn *thật* và mỗi hàng tương ứng với nhãn dự đoán của mô hình.
```

File: chapter_multilayer-perceptrons/environment_vn.md (hunk 10 above)
Comment (minhduc0711) [reply chapter_multilayer-perceptrons/environment_vn.md]: có dấu ngoặc ở cuối câu rồi mà :v

**Comment by nguyenduydu** [comment]
@lkhphuc  Cám ơn bạn đã góp ý.

File: chapter_multilayer-perceptrons/environment_vn.md [hunk 13]
```diff
@@ -699,7 +713,8 @@ với dấu `@` ở đầu. Ví dụ: @aivivn.
 *
 
 <!-- Phần 7 -->
-*
+* Nguyễn Duy Du 
+* Phạm Minh Đức
```
Comment (duythanhvn) [inline chapter_multilayer-perceptrons/environment_vn.md]:
```suggestion
* Phạm Minh Đức
* Lê Khắc Hồng Phúc
```
